File: app/src/main/java/com/worldventures/dreamtrips/modules/friends/storage/CirclesStorage.java
package com.worldventures.dreamtrips.modules.friends.storage;

import android.support.annotation.Nullable;

import com.worldventures.dreamtrips.core.janet.cache.CacheBundle;
import com.worldventures.dreamtrips.core.janet.cache.CachedAction;
import com.worldventures.dreamtrips.core.janet.cache.storage.ActionStorage;
import com.worldventures.dreamtrips.core.repository.SnappyRepository;
import com.worldventures.dreamtrips.modules.common.api.janet.command.CirclesCommand;
import com.worldventures.dreamtrips.modules.friends.model.Circle;

import java.util.List;

public class CirclesStorage implements ActionStorage<List<Circle>> {

   private final SnappyRepository db;

   public CirclesStorage(SnappyRepository db) {
      this.db = db;
   }

   @Override
   public Class<? extends CachedAction> getActionClass() {
      return CirclesCommand.class;
   }

   @Override
   public void save(@Nullable CacheBundle params, List<Circle> data) {
      db.saveCircles(data);
   }

   @Override
   public List<Circle> get(@Nullable CacheBundle action) {
      return db.getCircles();
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/users/circle/storage/CirclesStorage.kt
package com.worldventures.dreamtrips.social.service.users.circle.storage

import com.worldventures.janet.cache.CacheBundle
import com.worldventures.janet.cache.CachedAction
import com.worldventures.janet.cache.storage.ActionStorage
import com.worldventures.core.model.Circle
import com.worldventures.dreamtrips.social.domain.storage.SocialSnappyRepository
import com.worldventures.dreamtrips.social.service.users.circle.command.GetCirclesCommand

class CirclesStorage(private val db: SocialSnappyRepository) : ActionStorage<List<Circle>> {

   override fun getActionClass(): Class<out CachedAction<*>> = GetCirclesCommand::class.java

   override fun save(params: CacheBundle?, data: List<Circle>) = db.saveCircles(data)

   override fun get(action: CacheBundle?): List<Circle> = db.circles
}
